test/jb_funding_cycle_store/configure_for.test.js

```diff
@@ -0,0 +1,1396 @@
+    weight: ethers.BigNumber.from(0),
+    discountRate: ethers.BigNumber.from(0),
+    ballot: ethers.constants.AddressZero,
+    metadata: ethers.BigNumber.from(0)
+  };
+
+  const ballotStatus = {
+    APPROVED: 0,
+    ACTIVE: 1,
+    FAILED: 2
+  }
+
+  async function setup() {
+    const [deployer, controller, ...addrs] = await ethers.getSigners();
+
+    const mockJbDirectory = await deployMockContract(deployer, jbDirectory.abi);
+    const mockBallot = await deployMockContract(deployer, ijbFundingCycleBallot.abi);
+
+    const jbFundingCycleStoreFactory = await ethers.getContractFactory('JBFundingCycleStore');
+    const jbFundingCycleStore = await jbFundingCycleStoreFactory.deploy(mockJbDirectory.address);
+
+    return {
+      controller,
+      mockJbDirectory,
+      jbFundingCycleStore,
+      mockBallot,
+      addrs
+    };
+  }
+
+  const cleanFundingCycle = (fc) => ({
+    number: fc[0],
+    configuration: fc[1],
+    basedOn: fc[2],
+    start: fc[3],
+    duration: fc[4],
+    weight: fc[5],
+    discountRate: fc[6],
+    ballot: fc[7],
+    metadata: fc[8]
+  });
+
+  const fundingCycleMustStartOnOrAfterZero = ethers.BigNumber.from(0);
+
+  it("Should have no current or queued funding cycle before configuring", async function () {
+    const { jbFundingCycleStore } = await setup();
+
+    // Ballot status should be approved since there is no ballot.
+    expect(await jbFundingCycleStore.currentBallotStateOf(PROJECT_ID)).to.eql(0);
+    expect(cleanFundingCycle(await jbFundingCycleStore.currentOf(PROJECT_ID))).to.eql(EMPTY_FUNDING_CYCLE);
+    expect(cleanFundingCycle(await jbFundingCycleStore.queuedOf(PROJECT_ID))).to.eql(EMPTY_FUNDING_CYCLE);
+  });
+
+  it("Should create current funding cycle and queued cycle on first configure", async function () {
+    const { controller, mockJbDirectory, jbFundingCycleStore } = await setup();
+    await mockJbDirectory.mock.controllerOf.withArgs(PROJECT_ID).returns(controller.address);
+
+    const fundingCycleData = createFundingCycleData();
+
+    // The metadata value doesn't affect the test.
+    const fundingCycleMetadata = ethers.BigNumber.from(0);
+
+    // Configure funding cycle
+    const configureForTx = await jbFundingCycleStore
+      .connect(controller)
+      .configureFor(PROJECT_ID, fundingCycleData, fundingCycleMetadata, fundingCycleMustStartOnOrAfterZero);
+
+    // The timestamp the configuration was made during.
+    const configurationTimestamp = await getTimestamp(configureForTx.blockNumber);
+
+    await expect(configureForTx)
+      .to.emit(jbFundingCycleStore, 'Configure')
+      .withArgs(configurationTimestamp, PROJECT_ID, [fundingCycleData.duration, fundingCycleData.weight, fundingCycleData.discountRate, fundingCycleData.ballot], fundingCycleMetadata, fundingCycleMustStartOnOrAfterZero, controller.address);
+
+    await expect(configureForTx).to.emit(jbFundingCycleStore, `Init`)
+      .withArgs(configurationTimestamp, PROJECT_ID, /*basedOn=*/0);
```

Comment:
I see we're validating events in almost every test. Do we need to? It seems like we could have 1-2 tests that do this, but I feel the repetition of this (and other things) is detracting from being able to read exactly what the test is testing.

Reply:
I want to get `Init` covered in many different use cases. `Configure` is less important